include/type_safe/flag_set.hpp

```diff
@@ -416,6 +421,17 @@ class flag_set
     static_assert(flag_set_traits<Enum>::value, "invalid enum for flag_set");
 
 public:
+    /// \returns a flag_set based on the given integer value.
+    /// \requires `T` must be an unsigned integer type with number of bits <= internal used bit size.
+    template <typename T>
+    static constexpr flag_set from_int(T intVal)
+    {
+        static_assert(std::is_unsigned<T>::value
```

Comment:
To avoid any ambiguity, let's make it a `static_assert(std::is_same<T, int_type>::value)`. That way, you are less likely to accidentally forget about some bits (e.g. you pass an `uint32_t` but the actual enum is `uint64_t`).

Reply:
You are right, that makes sense. I will change it.

Reply:
Shall we make `using int_type = typename detail::flag_set_impl<Enum>::int_type;` public in flag_set? 
That way it could be used by the caller.

Reply:
Seems smart.

Reply:
I changed it.